spec/models/preserved_copy_spec.rb

```diff
@@ -309,4 +314,53 @@
       end
     end
   end
+
+  describe '#create_archive_preserved_copies!' do
```

Comment:
Add 2 tests:
1. ` expect { pc.create_archive_preserved_copies!(pc_version) }.to eq [ ArchivePreservedcopy object]`

2.  `create(:archive_preserved_copy)`
`expect { pc.create_archive_preserved_copies!(pc_version) }.to eq [] `

Reply:
So we expect that create_archive_preserved_copies returns the archive preserved copies we actually want, AND if we already have the archive preserved copies created, and we call the create_archive_preserved_copies, that method does not return any archive preserved copy objects.

Reply:
right, the method should only create the missing archive copies, and the method should return whatever copies it created.

so, e.g., if no copies were needed, an empty array should be returned.  if a new archive endpoint got added and connected to the relevant preservation policy, the method should create one new archive copy, and should return an instance of that archive copy.

Reply:
#985